### PR TITLE
fix(qwik-city): Prevent unexpected caching for `q-data.json` more

### PR DESCRIPTION
**Pull request by genki**
# What is it?

- [ ] Feature / enhancement
- [x] Bug
- [ ] Docs / tests / types / typos
- [ ] Infra

# Description

I have introduced the fetch options to avoid caching for `q-data.json` when `opts.cacheClear` is set (at #6797), but it was not enough.
As the `loadClientData` function manages caching by itself using `CLIENT_DATA_CACHE`, there is no need to use browser cache mechanism.
Or rather if they are used unexpectedly, it means we fetch not fresh `q-data.json` that may have been already changed in server side.
So I expanded the use of that options to the cases other than the `opts.cacheClear` is set, to all `GET` request as well.

# Checklist:

- [x] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my own code
- [ ] I have ran `pnpm change` and documented my changes
- [ ] I have made corresponding changes to the Qwik docs
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Review by wmertens — CHANGES_REQUESTED** [review]
Can you [add a changeset](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md#:~:text=Add%20a%20changeset%20with%20pnpm%20change%20if%20needed%20(follow%20this%20tutorial))?

So if I understand correctly, you should always deny caching because the client-side caching happens inside Qwik and not in the browser, right?

File: packages/qwik/src/optimizer/src/qwik-binding-map.ts [hunk 1]
```diff
@@ -30,15 +30,5 @@ export const QWIK_BINDING_MAP = {
         "platformArchABI": "qwik.win32-x64-msvc.node"
       }
     ]
-  },
```
[the remark this answers is not in the document]
Comment (genki) [reply packages/qwik/src/optimizer/src/qwik-binding-map.ts]: I did it. I don't know why this change has happened

**Comment by genki** [comment]
@wmertens I have added the changeset.
Yes, the fetched client data should be always fresh one if the cached one is missing in the Qwik City.

**Comment by genki** [comment]
I think if the contents of `q-data.json` is constant for all visitors, the server side caching is meaningful.
Browser cache is mostly no meaning but may be effective when the visitor is opening many tabs.
If there is a way to specify the scope that the q-data.json spatiotemporally being same value, we can put the scope id to the URL search params as the cache buster for invalidating the server side and browser cache. This is ideal.

**Comment by wmertens** [comment]
I've been thinking about this because of the discussion at https://github.com/QwikDev/qwik/issues/6789#issuecomment-2394982521. I looked into it but we can't know if a url is in the browser cache, and we can't know if it is being refetched.

So the prudent thing to do is indeed to disable caching on fetches. We need to come up with a more descriptive way to declare if routeLoaders are stale. Cache headers don't allow us to decide by search parameters, for example, and they force us to ask for each path separately.

**Review by wmertens — APPROVED** [review]
LGTM

**Comment by wmertens** [comment]
@genki I think this was wrong. We shouldn't override cache headers. Instead, if you need `q-data.json` to be fresh, you should fix your deployment so it serves the correct headers.

**Comment by wmertens** [comment]
@genki interesting, so the clearCache wasn't enough? BTW I think the stale-while-revalidate cache setting is really bad for q-data

**Comment by genki** [comment]
@wmertens 
Yes, you know the `clearCache` made navigation slow.
What I needed was the caching depending on some value that has lifetime same to the resource.
Here's an example that I had used the `cacheBuster`.

```tsx
<Link ... cacheBuster={paginate.value.updated_at.getTime().toString()}>
```

The framework couldn't extract such the value automatically, but the application programmer can tell it.

**Pull request merged**